pkg/segment/results/mresults/tests/metricsresults_test.go

```diff
@@ -617,3 +617,208 @@ func test_GetResults_Ops(t *testing.T, initialEntries map[uint32]float64, ansMap
+	}
+
+	test_GetResults_LogicalOps(t,
+		map[uint32]float64{
+			0:    45,
+			7200: 1045,
+		},
+		map[uint32]float64{
+			0:    2,
+			7200: 6,
+		},
+		results,
+		[]structs.QueryArithmetic{
+			{
+				LHS:       1,
+				RHS:       2,
+				Operation: utils.LetAnd,
+			},
+		},
+		structs.Downsampler{
+			Interval:   2,
+			Unit:       "h",
+			CFlag:      false,
+			Aggregator: structs.Aggreation{AggregatorFunction: utils.Avg},
+		},
+	)
+}
+
+func Test_GetResults_Or(t *testing.T) {
+	results := make(map[string]map[uint32]float64)
+	results["test.metric.1{color:red,type:compact,"] = map[uint32]float64{
+		0:    45,
+		7200: 1045,
+	}
+
+	results["test.metric.1{color:yellow,type:compact,"] = map[uint32]float64{
+		0:    45,
+		7200: 1045,
+	}
+
+	test_GetResults_LogicalOps(t,
+		map[uint32]float64{
+			0:    45,
+			7200: 1045,
+		},
+		map[uint32]float64{
+			0:    2,
+			7200: 6,
+		},
+		results,
+		[]structs.QueryArithmetic{
+			{
+				LHS:       1,
+				RHS:       2,
+				Operation: utils.LetOr,
+			},
+		},
+		structs.Downsampler{
+			Interval:   2,
+			Unit:       "h",
+			CFlag:      false,
+			Aggregator: structs.Aggreation{AggregatorFunction: utils.Avg},
+		},
+	)
+}
+
+func Test_GetResults_Unless(t *testing.T) {
+	results := make(map[string]map[uint32]float64)
+	results["test.metric.1{color:yellow,type:compact,"] = map[uint32]float64{
+		0:    45,
+		7200: 1045,
+	}
+
+	test_GetResults_LogicalOps(t,
+		map[uint32]float64{
+			0:    45,
+			7200: 1045,
+		},
+		map[uint32]float64{
+			0:    2,
+			7200: 6,
+		},
+		results,
+		[]structs.QueryArithmetic{
+			{
+				LHS:       1,
+				RHS:       2,
+				Operation: utils.LetUnless,
+			},
+		},
+		structs.Downsampler{
+			Interval:   2,
+			Unit:       "h",
+			CFlag:      false,
+			Aggregator: structs.Aggreation{AggregatorFunction: utils.Avg},
+		},
+	)
+}
+
+func initialize_Metric_Results(t *testing.T, initialEntries1 map[uint32]float64, initialEntries2 map[uint32]float64, queryOps []structs.QueryArithmetic, downsampler structs.Downsampler) map[uint64]*mresults.MetricsResult {
+	// Add 2 groups in metric1
+	mQuery1 := &structs.MetricsQuery{
+		MetricName:   "test.metric.1",
+		HashedMName:  1,
+		Downsampler:  downsampler,
+		GetAllLabels: true,
+	}
+	qid := uint64(0)
+	metricsResults1 := mresults.InitMetricResults(mQuery1, qid)
+	assert.NotNil(t, metricsResults1)
+
+	var tsGroupId1 *bytebufferpool.ByteBuffer = bytebufferpool.Get()
+	defer bytebufferpool.Put(tsGroupId1)
+	_, err := tsGroupId1.Write([]byte(mQuery1.MetricName + "{color:yellow,type:compact,"))
+	assert.NoError(t, err)
+
+	series := mresults.InitSeriesHolder(mQuery1, tsGroupId1)
+
+	for timestamp, val := range initialEntries1 {
+		series.AddEntry(timestamp, val)
+	}
+
+	metricsResults1.AddSeries(series, uint64(101), tsGroupId1)
+
+	var tsGroupId2 *bytebufferpool.ByteBuffer = bytebufferpool.Get()
+	defer bytebufferpool.Put(tsGroupId2)
+	_, err = tsGroupId2.Write([]byte(mQuery1.MetricName + "{color:red,type:compact,"))
```

Comment:
These spots with hardcoded strings: lines 728, 739, 752. Do you have those because in your tests (e.g., line 623) you have
```
results["test.metric.1{color:yellow,type:compact,"] = ...
```
I think that's going to cause problems in the future. If someone wants to write a test with different tags, how would that work?

Reply:
I will extract the tags